// See README.txt for information and build instructions.

package auxilo;

option java_package = "auxilo";
option java_outer_classname = "ProtobufMessages";

message HelloRequest
{
    required string customerID = 1;
    required string deviceName = 2;
    required bool isBox = 3;
}

//Runko kaikkien viestien valittamiseksi:
message Message
{
    //Kohdelaitteen tunniste viestin vastaanottajan maarittelemiseksi: Kohdelaitteen ID, tai ALL.
    required string deviceName = 1;
    optional DataMessage dataMesg = 2;	//Paluuviesti datan valittamiseksi TAI
    optional QueryMessage qry = 3;	//Varsinainen komento (laitteelle)

}

//Varsinainen laitteen tilasta kertova viesti:
message DataMessage
{
    required string hardwareID = 1;	//Anturin, toimilaitteen yksilollinen ID
    required int32 data = 2;		//Anturidata
    required string timeStamp = 3;
}

message QueryMessage
{
    required string sensorID = 1;		//Anturin tai laitteen ID
}

message DeviceList{

	repeated DeviceInfo deviceList = 1;
}

enum deviceType
{
    sensorTemp = 1;
    deviceNexa = 2;
    sensorPIR = 3;
};

message DeviceInfo
{
    required string boxID = 1;
    required string deviceID = 2;
    optional string description = 3;
    optional string aliasName = 4;

    required deviceType type = 5;
}


